improver/nbhood/recursive_filter.py

```diff
@@ -284,97 +281,100 @@ def _run_recursion(
-        elif smoothing_coefficients_cube.name() == "smoothing_coefficient_y":
-            smoothing_axis = "y"
-            non_smoothing_axis = "x"
-        else:
-            msg = (
-                "The smoothing coefficients cube must be named either "
-                "smoothing_coefficient_x or smoothing_coefficient_y. "
-                "The smoothing coefficients are named: "
-                f"{smoothing_coefficients_cube.name()}"
-            )
-            raise ValueError(msg)
-
-        smoothing_coord = smoothing_coefficients_cube.coord(axis=smoothing_axis)
-        non_smoothing_coord = smoothing_coefficients_cube.coord(axis=non_smoothing_axis)
-
-        mean_points = (
-            cube.coord(axis=smoothing_axis).points[1:]
-            + cube.coord(axis=smoothing_axis).points[:-1]
-        ) / 2
-
-        if len(smoothing_coord.points) != len(mean_points) or not np.allclose(
-            smoothing_coord.points, mean_points
-        ):
-            msg = (
-                f"The points of the {smoothing_axis} spatial dimension of the "
-                "smoothing coefficients must be equal to the mean of each pair "
-                f"of points along the {smoothing_axis} dimension of the input "
-                "cube."
-            )
-            raise ValueError(msg)
-
-        if len(non_smoothing_coord.points) != len(
-            cube.coord(axis=non_smoothing_axis).points
-        ) or not np.allclose(
-            non_smoothing_coord.points, cube.coord(axis=non_smoothing_axis).points
-        ):
-            msg = (
-                f"The points of the {non_smoothing_axis} spatial dimension of "
-                "the smoothing coefficients must be equal to the points along "
-                f"the {non_smoothing_axis} dimension of the input cube."
-            )
-            raise ValueError(msg)
+        # Ensure cubes are in x, y order.
+        smoothing_coefficients.sort(key=lambda cell: cell.name())
+        axes = ["x", "y"]
+
+        padded_coefficients = []
+        for axis, smoothing_coefficient in zip(axes, smoothing_coefficients):
+
+            # Check the smoothing coefficient cube name is as expected
+            expected_name = self.smoothing_coefficient_name_format.format(axis)
+            if smoothing_coefficient.name() != expected_name:
+                msg = (
+                    "The smoothing coefficient cube name {} does not match the "
+                    "expected name {}".format(
+                        smoothing_coefficient.name(), expected_name
+                    )
+                )
+                raise ValueError(msg)
 
-    def _set_smoothing_coefficients(self, smoothing_coefficients_cube):
-        """
-        Set up the smoothing_coefficient parameter.
+            # Check the smoothing coefficients do not exceed an empirically determined
+            # maximum value; larger values damage conservation significantly.
+            if (smoothing_coefficient.data > 0.5).any():
```

Comment:
This may cause problems if we choose to use the recursive filter in nowcast blending.  It's also inconsistent with the default maximum in the orographic alphas CLI / plugin, which doesn't have any controls on the maximum value it's allowed to generate.

However I see this is not new - so perhaps we should look at this separately.

Reply:
To not complicate this PR I suggest we do the following in something follow-on:
- move the ancillary_creation.py file from utilities into the generate_ancillaries directory (which would seem logical!).
- change the default values in the orographic alphas CLI / plugin to match our recommended maximum, though I don't think we should impose a limit there as we may want to use it experimentally.

Does that sound reasonable?

Reply:
Sounds reasonable to a point.  I think we should be imposing the maximum there, as these smoothing coefficients are designed for use in the recursive filter (and with nothing else), and the recursive filter imposes a maximum.  So we are imposing a limit already - it's just invisible to the user until they've wasted time on upstream steps.